core/block_dragger.js

```diff
@@ -307,6 +300,41 @@ Blockly.BlockDragger.prototype.endDrag = function(e, currentDragDeltaXY) {
   Blockly.Events.setGroup(false);
 };
 
+/**
+ * Calculates the drag delta and new location values after a block is dragged.
+ * If a block would not be moved because a drag target prevents block moves,
+ * then there is no delta, and the new location is the same as the startXY.
+ * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer has
+ *     moved from the start of the drag, in pixel units.
+ * @return {{delta: (!Blockly.utils.Coordinate|undefined), newLocation:
+ *     !Blockly.utils.Coordinate}} New location after drag. delta is in
+ *     workspace units, and is undefined if a move is prevented. newLocation is
+ *     the new coordinate where the block should end up.
+ * @protected
+ */
+Blockly.BlockDragger.prototype.getNewLocationAfterMove_ = function(
+    currentDragDeltaXY) {
+  var newValues = {};
+  var preventMove = !!this.dragTarget_ &&
+      this.dragTarget_.shouldPreventBlockMove(this.draggingBlock_);
+  if (preventMove) {
+    newValues.newLocation = this.startXY_;
+  } else {
+    newValues.delta = this.pixelsToWorkspaceUnits_(currentDragDeltaXY);
+    newValues.newLocation =
+        Blockly.utils.Coordinate.sum(this.startXY_, newValues.delta);
+  }
+
+  return newValues;
+};
+
+Blockly.BlockDragger.prototype.deleteBlockAfterMove_ = function() {
+  // Fire a move event, so we know where to go back to for an undo.
+  this.fireMoveEvent_();
+  this.draggingBlock_.dispose(false, true);
+  Blockly.draggingConnections = [];
```

Comment:
The logic in the dragger depends on the assumption that a block is only deleted if `wouldDeleteBlock_` is `true`.
Therefore, I think it'd be better to check that internal property to prevent a delete if `wouldDelete_` is `false`.

Reply:
We discussed offline and I changed it back similar to the old behavior of `maybeDeleteBlock`. This function handles the check and returns whether or not the deletion went through.